src/api/user/user.ts

```diff
@@ -0,0 +1,58 @@
+import { BadRequestError } from 'error-middleware/errors'
+import { validationMiddleware } from 'error-middleware/middlewares'
+import * as express from 'express'
+import * as asyncHandler from 'express-async-handler'
+import { UserModel } from '../../models/user'
+
+const { checkSchema, validationResult } = require('express-validator/check')
+
+const router = express.Router()
+
+router.get('/', (req, res) => {
+  res.send('Hello from user API')
+})
+
+const schema: any = {
+  email: {
+    isEmail: true,
+    in: 'body',
+    trim: true,
+    errorMessage: 'Invalid email',
+    custom: {
+      options: asyncHandler(async (value) => {
+        const user = await UserModel.getByEmail(value)
+        if (user) {
+          throw new BadRequestError('Email already exists')
```

Comment:
Do we need to throw an error inside of their custom validator?
Shouldn't we just return a value or smth?